pkg/asset/agent/agentconfig/agent_config.go

```diff
@@ -34,9 +40,49 @@ func (*Asset) Dependencies() []asset.Asset {
 	return []asset.Asset{}
 }
 
-// Generate generates the Agent manifest.
+// Generate generates the Agent Config manifest.
 func (a *Asset) Generate(dependencies asset.Parents) error {
-	return nil
+	agentConfig := &agent.Config{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "AgentConfig",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "example-agent-config",
+			Namespace: "cluster0",
+		},
+		Spec: agent.Spec{
+			RendezvousIP: "your-node0-ip",
+			Hosts: []agent.Host{
+				{
+					Hostname: "change-to-hostname",
```

Comment:
I have mixed feelings about whether we should fill in dummy values here, or just put empty fields. (The latter may mean we have to generate this as text directly, rather than Marshalling it from a struct.)
Either way, we will have to detect if we got the default config file and skip validation, and also not return any data from it from either `HostConfigFiles()` or any other way this asset's data might be accessed.
The implmentation of the latter might be more defensible if there are no values in the default.

Reply:
I've went with generate the text directly route. I don't know if we need to detect the template file. When "create cluster-manifests" is run, the installer should print out errors indicating there is an issue in the agent-config.yaml file. The user would correct and retry.